### DISCUSSION TITLE
Prisma Client - Select from junction table

### DISCUSSION THREAD

**Discussion by taylorhlabs** [prisma/prisma#4929] [Q&A] [answered]
I have read through the Prisma Client docs but I can't figure out how to get the syntax right for a query that involves a junction table. In my scenario, I have a Resource model and each Resource can be linked to 1 or more Tags in the Tag model. There is a junction table that joins them in my DB. How do I get all Tags for a given Resource when providing a Resource ID?

It's similar to getting the number of Posts for a User in the examples in the docs, but I have a junction table in this case.
```prisma
model Resource {
  id                      Int                       @id @default(autoincrement())
  userId                  Int                       @map("user_id")
  title                   String
  user                    User                      @relation(fields: [userId], references: [id])
  resourceTags            ResourcesTags[]

  @@index([userId], name: "resources_users_id_idx")
  @@map("resources")
}

model ResourcesTags {
  resourceId Int      @map("resource_id")
  tagId      Int      @map("tag_id")
  resource   Resource @relation(fields: [resourceId], references: [id])
  tag        Tag      @relation(fields: [tagId], references: [id])

  @@id([resourceId, tagId])
  @@index([tagId], name: "tag_resource_id")
  @@map("resources_tags_junction")
}

model Tag {
  id           Int                       @id @default(autoincrement())
  userId       Int                       @map("user_id")
  text         String
  deleted      Boolean                   @default(false)
  user         User                      @relation(fields: [userId], references: [id])
  tagResources ResourcesTags[]

  @@index([userId], name: "tags_users_id_idx")
  @@map("tags")
}
```

**Comment by ryands17** [comment, marked as answer]
Hey @taylorhlabs 👋 
Your query in this case would include the join table as follows:

```ts
await prisma.resource.findMany({
    include: { resourceTags: { include: { resource: true } } },
})
```

You can read more about that [here](https://www.prisma.io/docs/support/help-articles/working-with-many-to-many-relations)

Let me know if that works :)

**Reply by taylorhlabs** [reply]
Perfect, thanks @ryands17 for the code snippet. I never came across that article while scanning the docs but it's just what I needed.  🙏

**Reply by ryands17** [reply]
It's mentioned in the support articles [here](https://www.prisma.io/docs/support/help-articles/working-with-many-to-many-relations) :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
